### PR TITLE
Vejleder rettelser

### PR DESCRIPTION
**Pull request by mkju19**

### CODE REVIEW

File: sections/Chapter 5 Sprint 3/Sprint3.tex [hunk 1]
```diff
@@ -1,5 +1,17 @@
 \chapter{Sprint 3}
-%Common / Overall goal for all Knox groups
+This chapter covers the development process during the 3rd sprint. The 3rd sprint covers the the implementation of the needed database features for getting a functioning search engine.
```
Comment (Ivikhostrup) [inline sections/Chapter 5 Sprint 3/Sprint3.tex:2]:
```suggestion
This chapter covers the development process during the third sprint. The third sprint covers the implementation of the needed database features for getting a functioning search engine. 
```

File: sections/Chapter 5 Sprint 3/Sprint3.tex [hunk 2]
```diff
@@ -1,5 +1,17 @@
 \chapter{Sprint 3}
-%Common / Overall goal for all Knox groups
+This chapter covers the development process during the 3rd sprint. The 3rd sprint covers the the implementation of the needed database features for getting a functioning search engine. 
+
+
+The overall goal for the Knox project during this sprint was to get the search engine to work. To accomplish this goal the product owner specified a list of expectations for the entire \knox project, that when completed would fufill a minimal viable product. The specified list of common goals is outlined below.
```
Comment (Ivikhostrup) [inline sections/Chapter 5 Sprint 3/Sprint3.tex:5]:
```suggestion
To accomplish this goal, the product owner specified a list of expectations for the entire \knox{} project that, when completed, would fufill a minimal viable product. The specified list of common goals is outlined below.
```

File: sections/Chapter 5 Sprint 3/data-access.tex [hunk 3]
```diff
@@ -1,17 +1,22 @@
...
+\subsection{Object Relational Mapping}
+An Object Relational Mapping (ORM) is often used in an application to connect to, and manipulate data on a database.
+
+The primary requirement for the choice of an ORM was fast implementation and ease of use, due to the timeline constraints that were imposed by the \knox{} project. 
 
 \subsubsection{Dapper}
-Dapper is an open source Micro ORM or Micro object relational mapper. This means that Dapper is a mapping framework that helps mapping the native out queries to domain classes. Dapper is a light weight framework made for developers that prefers to use stored procedures or native query language over using a large scale ORM tool. 
+Dapper is an open source Micro ORM. This means that Dapper is a mapping framework that helps mapping the result from a native query to a class with the same attributes as the tables that are being queried. Dapper is a light weight framework made for developers that prefers to use stored procedures or native query language over using a large scale ORM tool.
```
Comment (Ivikhostrup) [inline sections/Chapter 5 Sprint 3/data-access.tex:7]:
```suggestion
Dapper is an open source Micro ORM. It is a mapping framework that helps mapping the result from a native query to a class with the same attributes as the tables. Dapper is a lightweight framework made for developers that emphasizes using stored procedures or native query language over using a large scale ORM tool. 
```

File: sections/Chapter 5 Sprint 3/data-access.tex [hunk 4]
```diff
@@ -1,17 +1,22 @@
...
+Dapper is an open source Micro ORM. This means that Dapper is a mapping framework that helps mapping the result from a native query to a class with the same attributes as the tables that are being queried. Dapper is a light weight framework made for developers that prefers to use stored procedures or native query language over using a large scale ORM tool. 
+
 
-Dapper, when added to projects, extendeds the IdbConnection inferface. This extension adds helpers for executing a command with no return results, executing a command multiple times, executing a query and mapping it to a strongly typed list and mapping it to a list of dynamic objects \cite{Dapper_Git}.
+Dapper, when added to projects, extends the IDbConnection inferface. This extension adds helpers. 
+One of the helpers is the Query() method. The query method will take an sql query as a parameter, the method returns a list. The type of list that is returned can either be a strongly typed list or a list of dynamic objects.
```
Comment (Ivikhostrup) [inline sections/Chapter 5 Sprint 3/data-access.tex:11]:
```suggestion
One of the helpers is the \texttt{Query} method. This method will take an SQL query as a parameter and return a list. The returned list can either be strongly or dynamically typed.
```

File: sections/Chapter 5 Sprint 3/Sprint3.tex (hunk 2 above)
Comment (Ivikhostrup) [reply sections/Chapter 5 Sprint 3/Sprint3.tex:5]: No need to repeat that this is the goal.

File: sections/Chapter 5 Sprint 3/Sprint3.tex [hunk 5]
```diff
@@ -1,5 +1,17 @@
...
+\begin{itemize}
+	\item It should be possible to make searches in all the available databases.
+	\item It should be possible to make searches using keywords in all the available data.
+	\item It should be possible to make searches in all Nordjyske articles from 2017-2021
+	\item It should be possible to make searches in all the Grundfos manuals, barring those that have shown to be problematic during processing.
+	\item The UI should minimally be able to display a link to the given article but preferably display the title and text.
```
Comment (Ivikhostrup) [inline sections/Chapter 5 Sprint 3/Sprint3.tex:11]:
```suggestion
	\item The UI should at least be able to display a link to the given article but preferably also display the title and text.
```

File: sections/Chapter 5 Sprint 3/data-access.tex [hunk 6]
```diff
@@ -1,17 +1,22 @@
-\subsection{Data access technologies}
-The primary requirement for the choice of technology was fast implementation and ease of use, due to the timeline constraints that were imposed by the \knox{} project. 
+\subsection{Object Relational Mapping}
+An Object Relational Mapping (ORM) is often used in an application to connect to, and manipulate data on a database.
```
Comment (Ivikhostrup) [inline sections/Chapter 5 Sprint 3/data-access.tex:2]:
```suggestion
An Object Relational Mapping (ORM) is often used in an application to connect to and manipulate data on a database.
```

File: sections/Chapter 5 Sprint 3/data-access.tex [hunk 7]
```diff
@@ -1,17 +1,22 @@
...
+
+The primary requirement for the choice of an ORM was fast implementation and ease of use, due to the timeline constraints that were imposed by the \knox{} project.
```
Comment (Ivikhostrup) [inline sections/Chapter 5 Sprint 3/data-access.tex:4]:
```suggestion
The primary requirement for the choice of an ORM was fast implementation and ease of use due to the timeline constraints that were imposed by the \knox{} project. 
```

File: sections/Chapter 5 Sprint 3/data-access.tex [hunk 8]
```diff
@@ -1,17 +1,22 @@
...
 
 \subsubsection{Dapper}
-Dapper is an open source Micro ORM or Micro object relational mapper. This means that Dapper is a mapping framework that helps mapping the native out queries to domain classes. Dapper is a light weight framework made for developers that prefers to use stored procedures or native query language over using a large scale ORM tool. 
+Dapper is an open source Micro ORM. This means that Dapper is a mapping framework that helps mapping the result from a native query to a class with the same attributes as the tables that are being queried. Dapper is a light weight framework made for developers that prefers to use stored procedures or native query language over using a large scale ORM tool. 
+
 
-Dapper, when added to projects, extendeds the IdbConnection inferface. This extension adds helpers for executing a command with no return results, executing a command multiple times, executing a query and mapping it to a strongly typed list and mapping it to a list of dynamic objects \cite{Dapper_Git}.
+Dapper, when added to projects, extends the IDbConnection inferface. This extension adds helpers.
```
Comment (Ivikhostrup) [inline sections/Chapter 5 Sprint 3/data-access.tex:10]:
```suggestion
Dapper, when added to projects, extends the \texttt{IDbConnection} inferface. This extension adds helper methods. 
```

**Review by Ivikhostrup — CHANGES_REQUESTED** [review]

File: sections/Chapter 5 Sprint 3/data-access.tex [hunk 9]
```diff
@@ -1,17 +1,22 @@
...
+
 
-Dapper, when added to projects, extendeds the IdbConnection inferface. This extension adds helpers for executing a command with no return results, executing a command multiple times, executing a query and mapping it to a strongly typed list and mapping it to a list of dynamic objects \cite{Dapper_Git}.
+Dapper, when added to projects, extends the IDbConnection inferface. This extension adds helpers. 
+One of the helpers is the Query() method. The query method will take an sql query as a parameter, the method returns a list. The type of list that is returned can either be a strongly typed list or a list of dynamic objects.
+Another helper that is added to IDbConnection is the Execute() method which takes some SQL commands as a parameter, and returns no results. When the method is called the commands used as parameter will be executed. The Execute() method can also be used to execute commands multiple times\cite{Dapper_Git}.
```
Comment (Pattrigue) [inline sections/Chapter 5 Sprint 3/data-access.tex:12]:
```suggestion
Another helper that is added to \texttt{IDbConnection} is the \texttt{Execute} method. This method takes one or more SQL commands as a parameter and has no return type. When called, the commands passed to the method will be executed. The \texttt{Execute} method can also be used to execute commands multiple times\cite{Dapper_Git}.
```

File: sections/Chapter 5 Sprint 3/data-access.tex [hunk 10]
```diff
@@ -1,17 +1,22 @@
...
 
 This means that Dapper as a framework covers our usecases with the caveat that additional time is going to be needed in order to fullfil the sprints MVP. 
 
 \subsubsection{Entity Framework Core}
-The second technology that we considered was Entity Framework Core (EF Core), which is a data access technology developed by Microsoft for the .NET platform. EF Core can serve as an object-relational mapper \cite{Object_relational_mapping} which lets developers abstract away the complications of converting data to .NET objects, in addition to eliminating the majority of the data-access code that would typically have to be written.
+The second technology that we considered was Entity Framework Core (EF Core), which is a data access technology developed by Microsoft for the .NET platform. EF Core can serve as an ORM \cite{Object_relational_mapping} which lets developers abstract away the complications of converting data to .NET objects, in addition to eliminating the majority of the data-access code that would typically have to be written.
```
Comment (Pattrigue) [inline sections/Chapter 5 Sprint 3/data-access.tex:19]:
```suggestion
The second technology that we considered was Entity Framework Core (EF Core), which is a data access technology developed by Microsoft for the .NET platform. EF Core can serve as an ORM \cite{Object_relational_mapping}, which lets developers abstract away the complications of converting data to .NET objects and eliminates the majority of the data-access code that would typically have to be written.
```

File: sections/Chapter 5 Sprint 3/planning.tex [hunk 11]
```diff
@@ -1,44 +1,35 @@
...
-\begin{itemize}
-	\item It should be possible to make searches in all the available databases.
-	\item It should be possible to make searches using keywords in all the available data.
-	\item It should be possible to make searches in all Nordjyske articles from 2017-2021
-	\item It should be possible to make searches in all the Grundfos manuals, barring those that have shown to be problematic during processing.
-	\item The UI should minimally be able to display a link to the given article but preferably display the title and text.
-\end{itemize}
+As noted in the preceding sprint, the codebase inherited from the previous iteration was deemed problematic, costly to maintain and did not have the required components in order for the pipeline to function. 
+Therefore our sprint goal for this iteration was based on rewriting codebase to comply with the requirements and adding the endpoints in order to get the pipeline working.
```
Comment (Pattrigue) [inline sections/Chapter 5 Sprint 3/planning.tex:3]:
```suggestion
Therefore our sprint goal for this iteration was to rewrite the codebase to comply with the requirements, and add the endpoints in order to get the pipeline working. 
```

File: sections/Chapter 5 Sprint 3/planning.tex [hunk 12]
```diff
@@ -1,44 +1,35 @@
...
 
-With the overall goal for the sprint outlined, a granulated sprint goal could be made that accounts for the focus of our scrum group.
 
-\subsubsection{Sprint goal}
-As noted in the preceding sprint, the codebase inherited from the previous iteration was deemed problematic and costly to maintain. 
-Therefore our sprint goal for this iteration was based on updating the codebase to better comply with the requirements for the \knox{} goal.
-
-
-Based on this observation as well as the aforementioned requirements, our sprint goal was as follows:
+Based on this observation as well as the aforementioned requirements, The MVP for our part of the sprint goal was as follows:
```
Comment (Pattrigue) [inline sections/Chapter 5 Sprint 3/planning.tex:6]:
```suggestion
Based on this observation as well as the aforementioned requirements, the MVP for our part of the sprint goal was as follows:
```

File: sections/Chapter 5 Sprint 3/planning.tex [hunk 13]
```diff
@@ -1,44 +1,35 @@
...
 \subsubsection{Backlog \& Increments}
-After defining the sprint goal, it was granulated into subtasks. 
-With the tasks granulated, a session of Scrum poker was held in order to evaluate the time requirement of each task. 
-Afterwards, the tasks were categorized by their priority and focus points.
-Lastly, the tasks were divided into three different sprint increments: \textit{Wordcount API}, \textit{Write Report}, \textit{Wiki Documentation}. 
-
-
-%%Remember to add an estimate to Write report and wiki update
-\begin{table}[]
+The next step of the planning phase is to granulate the sprint goal in to smaller task and then if possible further granulate these into subtasks.
```
Comment (Pattrigue) [inline sections/Chapter 5 Sprint 3/planning.tex:12]:
```suggestion
The next step of the planning phase was to granulate the sprint goal into smaller tasks, and further granulate these into subtasks if possible.
```

File: sections/Chapter 5 Sprint 3/planning.tex [hunk 14]
```diff
@@ -1,44 +1,35 @@
...
+                            & Delete   & 3          &  &  \\ \cline{1-3}
+Docker                      &          & 13         &  &  \\ \cline{1-3}
+Repository Service          &          & 10         &  &  \\ \cline{1-3}
+CI/CD                       &          & 5          &  &  \\ \cline{1-3}
 \end{tabular}
+\caption{Backlog Granulation \& Estimation}
+\label{BacklogEstimationSprint3}
 \end{table}
 
+With the planning phase of the sprint completed the development phase can now begin.
```
Comment (Pattrigue) [inline sections/Chapter 5 Sprint 3/planning.tex:32]:
```suggestion
With the planning phase of the sprint completed, the development phase can now begin. 
```

**Review by Ivikhostrup — APPROVED** [review]

**Pull request merged**